app/models/drupal_node_revision.rb

```diff
@@ -43,6 +43,13 @@ def inline_tags
     end
   end
 
+  # search for inline hashtags(such as #hashtag) and create a new tag
+  def inline_hashtags
+    self.body.scan(/[^\n]\#([\w-]+)/).each do |match|
```

Comment:
That makes sense, but I think that regex also doesn't work for things such as this: 
```
#Heading 1
Here is a #hashtag
#Heading 2
Here is another #hashtag
```
that regex will match heading 2 if I'm not mistaken. Let me know if you have an idea of how to avoid markdown headers.  Thanks!